src/main/java/online/partyrun/partyrunmatchingservice/global/sse/MultiSinkHandler.java

```diff
@@ -0,0 +1,86 @@
+package online.partyrun.partyrunmatchingservice.global.sse;
+
+import lombok.AccessLevel;
+import lombok.experimental.FieldDefaults;
+
+import online.partyrun.partyrunmatchingservice.global.sse.exception.InvalidSinksKeyException;
+import online.partyrun.partyrunmatchingservice.global.sse.exception.SseConnectionException;
+
+import org.springframework.stereotype.Component;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Sinks;
+import reactor.core.scheduler.Schedulers;
+
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+@Component
+@FieldDefaults(level = AccessLevel.PRIVATE, makeFinal = true)
+public class MultiSinkHandler<K, V> implements ServerSentEventHandler<K, V> {
+    Map<K, Sinks.Many<V>> sinks = new HashMap<>();
+
+    @Override
+    public Flux<V> connect(final K key) {
+        return getSink(key)
+                .asFlux()
+                .subscribeOn(Schedulers.boundedElastic())
+                .timeout(timeout())
+                .doOnCancel(() -> sinks.remove(key))
+                .doOnError(
+                        e -> {
+                            sinks.remove(key);
+                            throw new SseConnectionException();
+                        });
+    }
+
+    @Override
+    public void sendEvent(final K key, final V value) {
+        getSink(key).tryEmitNext(value);
+    }
+
+    @Override
+    public void complete(final K key) {
+        getSink(key).tryEmitComplete();
+        sinks.remove(key);
+    }
+
+    @Override
+    public void create(K key) {
+        validateKey(key);
+        sinks.putIfAbsent(key, Sinks.many().replay().all());
```

Comment:
```suggestion
        sinks.putIfAbsent(key, Sinks.many().unicast().onBackpressureBuffer());
```

sink가 unicat로 진행되니까 이렇게 바꿔도 좋을 것 같아요.